base/src/QRReader.cpp

```diff
@@ -123,11 +143,34 @@ bool QRReader::term()
 bool QRReader::process(frame_container &frames)
 {
 	auto frame = frames.begin()->second;
-
-	const auto &result = ZXing::ReadBarcode({static_cast<uint8_t *>(frame->data()), mDetail->mWidth, mDetail->mHeight, mDetail->mImageFormat}, mDetail->mHints);
+ 
+	const auto &result = ZXing::ReadBarcode({static_cast<uint8_t *>(frame->data()), mDetail->mWidth, mDetail->mHeight, mDetail->mImageFormat}, mDetail->mReaderOptions);
 	
-	auto text = ZXing::TextUtfEncoding::ToUtf8(result.text());
+	auto text = result.text();
+	if(text.length())
+	{
+		LOG_INFO << "ZXING decoded QR: " << text;
+	}
 	
+	if (mDetail->mSaveQRImages && (mDetail->mQRImagesFolderName != ""))
+	{
+		fs::path savePath = mDetail->mQRImagesFolderName / (std::to_string(mDetail->mFrameCounter) + ".raw");
+        std::ofstream outFile(savePath.string(), std::ios::binary);
+        if (outFile)
+        {
+            outFile.write(static_cast<char*>(frame->data()), frame->size());
+            outFile.close();
+        }
+        else
+        {
+            LOG_ERROR << "Failed to save frame to " << savePath.string();
+        }
```

Comment:
this block is poorly indented... and it confuses the reviewer. In general just using the built in code formatter in VScode/VS is enough here

Reply:
also if the savePath is not a good directly... we will be gettting an std::exception thrown... which is not handled here

Reply:
Now I have indented properly and also added trycatch block to handle exception